File: python/batch_get_last_reported_points.py
from client import create_native_client
import nativeconnect.api.v1.nativeconnect_pb2 as native


def main():
    client = create_native_client()

    request = native.BatchGetLastReportedPointsRequest(
        vins=["LRW3F7FS9PC893577", "1C4RJHKG2P8863343"],
        points=5
    )

    response = client.BatchGetLastReportedPoints(request)

    print(response)


if __name__ == "__main__":
    main()

File: typescript/src/batch_get_last_reported_points.ts
import { createNodeClient } from "./client"
import { create, toJsonString } from "@bufbuild/protobuf";
import {
  BatchGetLastReportedPointsRequestSchema,
  BatchGetLastReportedPointsResponseSchema
} from "@buf/nativeconnect_api.bufbuild_es/nativeconnect/api/v1/nativeconnect_pb";

const client = createNodeClient()

const request = create(BatchGetLastReportedPointsRequestSchema, {
  vins: ["LRW3F7FS9PC893577", "1C4RJHKG2P8863343"],
  points: 5
})

const response = await client.batchGetLastReportedPoints(request)

console.log(toJsonString(BatchGetLastReportedPointsResponseSchema, response, { prettySpaces: 2 }))
